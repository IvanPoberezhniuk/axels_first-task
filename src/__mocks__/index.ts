import { Comment, Image, ImageDetails } from '../redux/ducks/images';

const image: Image = {
  id: 222,
  url: '/fakeUri/images'
};

const comment: Comment = {
  id: 222,
  text: 'test',
  date: new Date().getSeconds()
};

const imagesDetails: ImageDetails = {
  id: 222,
  url: '/fakeUri/images',
  comments: [{
    id: 222,
    text: 'test',
    date: new Date().getSeconds()
  }, {
    id: 222,
    text: 'test',
    date: new Date().getSeconds()
  }]
};
