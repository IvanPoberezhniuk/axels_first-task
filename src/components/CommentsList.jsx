import React from 'react';
import PropTypes from 'prop-types';

import {
  CommentListGroup,
  CommentListItem,
  CommentListParagraph,
  CommentListTime
} from '../styled/components/CommentsList';

const CommentsList = ({comments}) => {
  const convertDate = (comment) => {
    return new Date(comment.date).toLocaleDateString().split('/').join('.');
  };

  return (
    <CommentListGroup>
      {comments.map(comment => (
        <CommentListItem key={comment.id}>
          <CommentListParagraph>
            <CommentListTime> {convertDate(comment)} </CommentListTime>
          </CommentListParagraph>
          <CommentListParagraph>{comment.text}</CommentListParagraph>
        </CommentListItem>))}
    </CommentListGroup>
  );
};

CommentsList.propTypes = {
  comments: PropTypes.arrayOf(PropTypes.shape({
    date: PropTypes.number,
    id: PropTypes.number,
    text: PropTypes.string
  })).isRequired
};

export default CommentsList;
