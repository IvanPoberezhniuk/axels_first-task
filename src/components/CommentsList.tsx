import React from 'react';

import {
  CommentListGroup,
  CommentListItem,
  CommentListParagraph,
  CommentListTime
} from '../styled/components/CommentsList';

interface Comment {
  id: number,
  text: string,
  date: number
}

interface Props {
  comments: Array<Comment>
}

const CommentsList = ({ comments }: Props) => {
  const convertDate = (comment: Comment) =>
    new Date(comment.date).toLocaleDateString().split('/').join('.');

  return (
    <CommentListGroup>
      {comments.map((comment) => (
        <CommentListItem key={comment.id}>
          <CommentListParagraph>
            <CommentListTime> {convertDate(comment)} </CommentListTime>
          </CommentListParagraph>
          <CommentListParagraph>{comment.text}</CommentListParagraph>
        </CommentListItem>
      ))}
    </CommentListGroup>
  );
};

export default CommentsList;
