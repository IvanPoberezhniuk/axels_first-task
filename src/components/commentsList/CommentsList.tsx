import React from 'react';

import {
  CommentListGroup,
  CommentListItem,
  CommentListParagraph,
  CommentListTime
} from '../../styled/components/CommentsList';
import { convertDate } from '../../utils/date';
import { Comment } from '../../redux/ducks/images';

export interface Props {
  comments?: Array<Comment>;
}

const CommentsList = ({ comments = [] }: Props) => {
  return (
    <CommentListGroup>
      {comments.map((comment: Comment) => (
        <CommentListItem key={comment.id}>
          <CommentListParagraph>
            <CommentListTime> {convertDate(comment)} </CommentListTime>
          </CommentListParagraph>
          <CommentListParagraph>{comment.text}</CommentListParagraph>
        </CommentListItem>
      ))}
    </CommentListGroup>
  );
};

export default CommentsList;
