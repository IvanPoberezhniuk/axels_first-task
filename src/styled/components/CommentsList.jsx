import styled from 'styled-components';
import {ListGroup} from 'react-bootstrap';

const CommentListParagraph = styled.p`
  margin: 0;
`;
const CommentListTime = styled.time`
  color: gray;
`;

const CommentListItem = styled(ListGroup.Item)`
  border: none;
  padding: 0 0 1rem 0;
`;

const CommentListGroup = styled(ListGroup)`
  overflow-y: scroll;
  overflow-x: hidden;

  ::-webkit-scrollbar {
    width: 0px; /* Remove scrollbar space */
    background: transparent; /* Optional: just make scrollbar invisible */
  }
`;

export {CommentListParagraph, CommentListItem, CommentListGroup, CommentListTime};
