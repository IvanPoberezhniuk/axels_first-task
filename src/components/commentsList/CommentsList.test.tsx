import { shallow, ShallowWrapper } from 'enzyme';

import { CommentsList } from '../index';

import { Props } from './CommentsList';
import { mockedComment } from '../../__mocks__';

let component: ShallowWrapper;
const setUp = (props?: Props) => shallow(<CommentsList {...props} />);

beforeEach(() => {
  component = setUp({ comments: [mockedComment, mockedComment] });
});

describe('CommentsList component', () => {
  it('SNAPSHOT', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render 2 comments', () => {
    const comments = component.find('Styled(ListGroupItem)');
    expect(comments.length).toBe(2);
  });

  it('should render 2 comments', () => {
    component = setUp();

    const comments = component.find('Styled(ListGroupItem)');
    expect(comments.length).toBe(0);
  });
});

