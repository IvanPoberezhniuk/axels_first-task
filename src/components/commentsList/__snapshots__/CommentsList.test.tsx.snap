// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsList component SNAPSHOT 1`] = `
<Styled(ListGroup)>
  <Styled(ListGroupItem)
    key="222"
  >
    <styled.p>
      <styled.time>
         
        1.1.1970
         
      </styled.time>
    </styled.p>
    <styled.p>
      test
    </styled.p>
  </Styled(ListGroupItem)>
  <Styled(ListGroupItem)
    key="222"
  >
    <styled.p>
      <styled.time>
         
        1.1.1970
         
      </styled.time>
    </styled.p>
    <styled.p>
      test
    </styled.p>
  </Styled(ListGroupItem)>
</Styled(ListGroup)>
`;
